juju/juju
# fix(migration): move description model validation after domain

Currently, after filling the `description.Model` when migrating the legacy db contents we validate it. This happens before we start filling it with the contents from dqlite.

We face an issue here, that on some cases the validation will fail because some of the missing data will arrive after we finish the dqlite stage of the migration. This is the case for machines/instance data migration.

We therefore move the whole Validation() to internal/migration, *after* dqlite.





## Checklist




## QA steps

Migration should not be affected:

```
juju bootstrap localhost tgt
juju bootstrap localhost src
juju add-model m
juju deploy ubuntu
# Wait until ACTIVE
juju migrate m tgt
```
Also, https://github.com/juju/juju/pull/18214 should pass the migration checks after rebased on top of this one.


## Links


**Jira card:** JUJU-
